### ISSUE TITLE
Ambiguous proof search candidates lead to default hints

### ISSUE THREAD

**Issue by ohad** [edwinb/Idris2-boot#162]
Not sure whether this is a bug or intended behaviour.

# Steps to Reproduce
```
data Gnu : Type where [uniqueSearch,noHints]
  G : Bool -> Gnu
  N : Bool -> Gnu
  U : Bool -> Gnu

||| Makes interaction a bit nicer
Show Gnu where
  show (G b) = "G: " ++ show b
  show (N b) = "N: " ++ show b
  show (U b) = "U: " ++ show b
  
%defaulthint
FooG : Gnu
FooG = G False

FooN : Gnu
FooN = N False

%hint
FooU : Gnu
FooU = U False

||| Curried versions
CooG : Unit -> Gnu
CooG _ = FooG

%hint
CooN : Unit -> Gnu
CooN _ = FooN

CooU : Unit -> Gnu
CooU _ = FooU

||| We'll use this function to see what proof Idris finds.
Guess : {auto gnu : Gnu} -> String
Guess {gnu} = show gnu
  
A : String
A = Guess
```

# Expected Behavior
```
Foo.idr:39:5--40:1:While processing right hand side of Main.A at Foo.idr:39:1--40:1:
Multiple solutions found in search. Possible correct results:
	CooN ()
	FooU
```

# Observed Behavior
```
Main> A
"G: False"
```

I think this is because the proof search starts off with default hints disabled, finds an ambiguity and fails, and because search without defaults failed, we resume with defaults enabled, and we find `FooG`.

This is either wrong, or should be explicitly document as a design choice, as it leads to silent failure. Since default arguments exclude ambiguity, it's more consistent to consider this as a bug, and require that in this case Idris should report an ambiguity error.

**Comment by ohad** [comment]
Fixed in 036bb851345e4e719b5d750666e73abed0a933a6 on my branch.
Added as a regression test in db1d69d34ef9c118d4504eb6c32c55e351987a3a on my branch.
I want to try and fix #160 before creating a pull request.

**Comment by edwinb** [comment]
I think I fixed this one in passing, it certainly looks like another issue I encountered with default hints recently. And if it's fixed for you too, I'll close.
